python/tk_desktop/rpc.py

```diff
@@ -280,10 +287,24 @@ def run(self):
                     ready = False
             else:
                 # can use select on osx and linux
-                (rd, _, _) = select.select(
-                    [self.server._listener._socket], [], [], self.LISTEN_TIMEOUT
-                )
-                ready = len(rd) > 0
+                try:
+                    (rd, _, _) = select.select(
+                        [self.server._listener._socket], [], [], self.LISTEN_TIMEOUT
+                    )
+                    ready = len(rd) > 0
+                except AttributeError as e:
+                    # The server has been closed. `_listener` is None.
+                    logger.debug("Error during select:", exc_info=True)
+                    if self.server._listener:
+                        raise
+                    ready = False
+                except select.error as e:
+                    import errno
```

Comment:
I always wonder about those. Should we import everything at the beginning of the file or at "runtime"?

Reply:
I think I'm going to stick to PEP 8. We don't have a circular import problem here, and the import is not expensive, so I don't see why not put it at the beginning of the file. Moving it.